apps/os/backend/agent/context.ts

```diff
@@ -347,26 +348,107 @@ function parseHm(hhmm: string): number {
+    }
+  }
+  
+  if (endIndex === -1) {
+    return { frontMatter: {}, body: content };
+  }
+  
+  const frontMatterText = lines.slice(1, endIndex).join("\n");
+  const body = lines.slice(endIndex + 1).join("\n").trim();
+  
+  try {
+    const frontMatter = parseYaml(frontMatterText) as Record<string, unknown>;
+    return { frontMatter: frontMatter || {}, body };
+  } catch (error) {
+    logger.warn(`Failed to parse front matter as YAML:`, error);
+    return { frontMatter: {}, body: content };
+  }
+}
+
+/**
+ * Converts a match value from front matter into a proper ContextRuleMatcher.
+ * Handles two cases:
+ * 1. If match is a string, treat it as a jsonata expression (JSON data matcher)
+ * 2. If match is an object, treat it as a ContextRuleMatcher directly
+ */
+export function parseFrontMatterMatch(match: unknown): ContextRuleMatcher | undefined {
+  if (match === undefined || match === null) {
+    return undefined;
+  }
+  
+  if (typeof match === "string") {
+    return { type: "jsonata", expression: match };
+  }
+  
+  return match as ContextRuleMatcher;
+}
+
 /**
  * Helper function to create context rules from files matching a glob pattern.
  * Each file becomes a context rule with slug derived from filename and prompt from file content.
+ * Supports YAML front matter for overriding context rule properties.
  */
 export function contextRulesFromFiles(pattern: string, overrides: Partial<ContextRule> = {}) {
   try {
-    // Resolve pattern relative to the config file's directory
     const configDir = findIterateConfig();
     if (!configDir) {
       throw new Error("iterate.config.ts not found");
     }
     const files = globSync(pattern, { cwd: configDir }) as string[];
     return files.map((filePath: string) => {
       const fileContent = readFileSync(join(configDir, filePath), "utf-8");
-      // Get relative path from config directory and remove .md extension
-      return defineRule({
-        key: filePath.replace(/\.md$/, ""),
-        prompt: fileContent,
+      const { frontMatter, body } = parseFrontMatter(fileContent);
+      
+      const defaultKey = filePath.replace(/\.md$/, "");
+      
+      const rule: ContextRule = {
+        key: defaultKey,
+        prompt: body,
         ...overrides,
-      });
+      };
+      
+      if (frontMatter.key && typeof frontMatter.key === "string") {
+        rule.key = frontMatter.key;
+      }
+      
+      if (frontMatter.slug && typeof frontMatter.slug === "string") {
+        rule.key = frontMatter.slug;
+      }
+      
+      if (frontMatter.description && typeof frontMatter.description === "string") {
+        rule.description = frontMatter.description;
+      }
+      
+      if (frontMatter.match !== undefined) {
+        rule.match = parseFrontMatterMatch(frontMatter.match);
+      }
+      
+      if (frontMatter.tools !== undefined) {
+        rule.tools = frontMatter.tools as ContextRule["tools"];
+      }
+      
+      return defineRule(rule);
```

Comment:
```suggestion
      return defineRule({
        key: defaultKey,
        prompt: body,
        match: parseFrontMatterMatch(match),
        ...otherFrontMatter
      });
```

ithink we can just do this no?